src/main/java/org/osc/controller/nsc/api/SampleSdnRedirectionApi.java

```diff
@@ -495,4 +500,135 @@ public void close() throws Exception {
         });
     }
 
+    // Support methods for REST API's
+
+    public List<String> getInspectionHooksIds() throws Exception {
+
+        return this.txControl.supports(() -> {
+
+            List<InspectionHookEntity> elements = this.utils.txInspectionHookEntities();
+            List<String> elementList = new ArrayList<String>();
+
+            if (!elements.isEmpty()) {
+                elementList = elements.stream().map(InspectionHookEntity::getHookId).collect(Collectors.toList());
+            }
+
+            return elementList;
+        });
+    }
+
+    public Element updateInspectionPort(InspectionPortElement inspectionPort) throws Exception {
+
+        InspectionPortEntity inspectionPortEntity = (InspectionPortEntity) getInspectionPort(inspectionPort);
+        if (inspectionPortEntity == null) {
```

Comment:
use org.osc.controller.nsc.utils.RedirectionApiUtils.throwExceptionIfNullElement(InspectionPortElement)

Reply:
@arvindn05 , these are private methods and will make it public.